### PR TITLE
chore: Add new SDK overhead page for Apple platforms

### PR DESCRIPTION
**Pull request by kahest**
## DESCRIBE YOUR PR

Adds new "SDK Overhead" menu item + page below the "Mobile SDK Releases" item. Covers Errors etc. in more detail, links to existing product pages for Performance+Profiling+SR.

follow-up to https://github.com/getsentry/sentry-docs/pull/12031

fixes https://github.com/getsentry/projects/issues/453

Next steps/future work: see https://github.com/getsentry/projects/issues/169

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [x] Other deadline: no specific date, target is Dec 11
- [ ] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: docs/platforms/apple/common/overhead/index.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+As stated in our <Link to="/product/explore/profiling/performance-overhead">product docs on the topic</Link>, Profiling adds some overhead, but should have minimal impact on the performance of your application. In typical scenarios, the expected overhead is less than 5% of the app's resource utilization.
+
+</PlatformSection>
+
+<PlatformSection supported={["apple.ios"]}>
+
+## Session Replay
+
+As stated in our <Link to="/product/explore/session-replay/mobile/performance-overhead">product docs on the topic</Link>, Session Replay adds some overhead, but should have minimal impact on the performance of your application. For more details on the measured overhead, read the <PlatformLink to="/session-replay/performance-overhead/">performance overhead docs for this SDK</PlatformLink>.
```
Comment (philipphofmann) [inline docs/platforms/apple/common/overhead/index.mdx:57]: Should we mention that every time we capture a screenshot you get a slightly delayed slow frame? Having it turned on causes plenty of slow frames for your UIViewController transactions, for example.

**Review by philipphofmann — COMMENTED** [review]
A few minor comments.

File: docs/platforms/apple/common/overhead/index.mdx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+<PlatformLink to="/enriching-events/breadcrumbs">Breadcrumbs</PlatformLink> are collected through automated integrations or by manually adding them. To have them readily available for every event generated by the SDK, they are continuously persisted, and managed in a performant buffer. This shouldn't impact user experience.
+
+Capturing excessive numbers of breadcrumbs (for example, creating breadcrumbs for all log messages) can cause significant performance overhead. To mitigate this, review and adapt your app's usage of breadcrumbs. For example, increase the min-level of log messages that create breadcrumbs from `warn` to `error`.
+
+Note that increasing the max number of breadcrumbs **does not** improve performance and can even have a detrimental effect.
+
+## Tracing and Performance Monitoring
+
+As stated in our <Link to="/product/performance/performance-overhead">product docs on the topic</Link>, Tracing adds some overhead, but should have minimal impact on the performance of your application. In typical scenarios, the expected overhead is less than 3% of the app's resource utilization.
```
[the remark this answers is not in the document]
Comment (kahest) [reply docs/platforms/apple/common/overhead/index.mdx:43]: it's a rough estimation/guideline of the upper limit, hence we say "less than". we'll iterate soon with more concrete data from benchmarks

File: docs/platforms/apple/common/overhead/index.mdx (hunk 1 above)
Comment (kahest) [reply docs/platforms/apple/common/overhead/index.mdx:57]: I think this is mainly true for low end iOS devices, right? or in general?

might be something to (also) put on the SR perf overhead docs for SR. @brustolin @romtsn wdyt?

Comment (brustolin) [reply docs/platforms/apple/common/overhead/index.mdx:57]: Yes, I think it should be mentioned.

Comment (romtsn) [reply docs/platforms/apple/common/overhead/index.mdx:57]: I mean we kind of put the numbers there that imply that you'll have a slow frame, but yeah we could mention that explicitly:
image

Comment (kahest) [reply docs/platforms/apple/common/overhead/index.mdx:57]: added a note below -> https://github.com/getsentry/sentry-docs/pull/12063/files#diff-47891e19e6939b70d674d775a7d082e3188f455a519dbab3f3cee073b35e7977R59

File: docs/platforms/apple/common/overhead/index.mdx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+---
+title: SDK Overhead
+description: "Learn about Sentry's Apple SDK overhead."
```
Comment (coolguyzone) [inline docs/platforms/apple/common/overhead/index.mdx:3]: Could you make this a little more descriptive? Perhaps something like:

```suggestion
description: "Learn about Sentry's Apple SDK overhead and how you can tailor your configuration to minimize it."
```

**Review by coolguyzone — APPROVED** [review]
Thanks for writing this up, looks good to me! 🤙

File: docs/platforms/apple/common/overhead/index.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (coolguyzone) [reply docs/platforms/apple/common/overhead/index.mdx:43]: I recommend we be extra generous when making estimates like this. There are liability issues if we cut it close and end up printing a statistic that turns out to be incorrect.

**Pull request merged**